src/test/java/io/airlift/units/TestDataSizeValidator.java

```diff
@@ -83,16 +88,15 @@ public void testDetectsBrokenMinAnnotation()
         assertThatThrownBy(() -> VALIDATOR.validate(new BrokenMinAnnotation()))
                 .isInstanceOf(ValidationException.class)
                 .hasRootCauseInstanceOf(IllegalArgumentException.class)
-                .hasMessage("java.lang.IllegalArgumentException: size is not a valid data size string: broken");
+                .hasStackTraceContaining("java.lang.IllegalArgumentException: size is not a valid data size string: broken");
```

Comment:
keep testing with `hasMessage`

optionally add additional assertions with hasStackTraceContaining

Reply:
Done. I've found one bug in a test